### PR TITLE
SONARJAVA-1138 Update S1186 to include no-arg constructor

### PR DESCRIPTION
**Pull request by mpaladin**
http://jira.sonarsource.com/browse/SONARJAVA-1138

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/helpers/SyntaxNodePredicates.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import org.sonar.plugins.java.api.tree.Tree;
+
+public class SyntaxNodePredicates {
+
+  private SyntaxNodePredicates() {
+    // Useful predicates to be used with syntax nodes
+  }
+
+  public static Predicate<Tree> kind(final Tree.Kind syntaxNodeKind) {
+    return new Predicate<Tree>() {
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/helpers/SyntaxNodePredicates.java]: MAJOR Make this a named "static" inner class. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2694)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: java-checks/src/main/java/org/sonar/java/checks/helpers/SyntaxNodePredicates.java (hunk 1 above)
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/helpers/SyntaxNodePredicates.java]: This one is a boring FP... We should create a ticket for it.

File: java-checks/src/main/java/org/sonar/java/checks/EmptyMethodsCheck.java [hunk 2]
```diff
@@ -57,30 +59,27 @@ public void visitNode(Tree tree) {
...
-        if (member.is(Kind.METHOD)) {
+        if (member.is(Kind.METHOD) || isPublicConstructor(member)) {
           checkMethod((MethodTree) member);
         }
       }
     }
   }
 
+  private static boolean isPublicConstructor(Tree node) {
+    return node.is(Kind.CONSTRUCTOR) && ModifiersUtils.hasModifier(((MethodTree) node).modifiers(), Modifier.PUBLIC);
```
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/EmptyMethodsCheck.java]: The jira ticket is about no-arg constructors but here you only look at public constructors without looking at arguments. Is it expected?

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Pull request merged**